### DISCUSSION TITLE
Linking distributed spans with `tracing-opentelemetry` and `opentelemetry-jaeger`

### DISCUSSION THREAD

**Discussion by mempirate** [tokio-rs/tracing#2439] [Q&A] [answered]
I'm working on integrating `tracing-opentelemetry` in a distributed system that uses a Jaeger subscriber. I have messages propagating through the system that have a unique ID, and each system does some work with that message. How can I link these spans together using the unique ID with `tracing-opentelemetry`? I thought about setting the trace ID, but according to https://docs.rs/tracing-opentelemetry/latest/tracing_opentelemetry/index.html#special-fields, that's not a name we can set.

**Comment by davidbarsky** [comment, marked as answer]
Generally speaking, we recommend using this extension trait to propagate non-`tracing` context: https://docs.rs/tracing-opentelemetry/latest/tracing_opentelemetry/trait.OpenTelemetrySpanExt.html

**Reply by mempirate** [reply]
Figured it out! Thanks :)
